ledger/ledger-on-memory/src/main/scala/com/daml/ledger/on/memory/InMemoryState.scala

```diff
@@ -16,17 +17,22 @@ import scala.concurrent.{ExecutionContext, Future}
 
 private[memory] class InMemoryState private (log: MutableLog, state: MutableState) {
   private val lockCurrentState = new Semaphore(1, true)
+  private val lastLogEntry = new AtomicInteger()
 
   def readLog[A](action: ImmutableLog => A): A =
     action(log) // `log` is mutable, but the interface is immutable
 
+  def logSize(): Int = lastLogEntry.get()
+
   def write[A](action: (MutableLog, MutableState) => Future[A])(
       implicit executionContext: ExecutionContext
   ): Future[A] = {
     lockCurrentState.acquire()
     action(log, state)
       .andThen {
-        case _ => lockCurrentState.release()
+        case _ =>
+          lastLogEntry.set(log.size)
```

Comment:
The previous semantics were setting the dispatcher's new head to the last index (see `appendEntry`); it gets the size _before_ appending. This sets it to the size _after_ appending. You'll either need to redo everything to make sure this is consistent, or set this to `log.size - 1`.

Reply:
Good catch -- changed accordingly.